File: src/main/java/com/smart/appsa/model/enums/TipoPedido.java
package com.smart.appsa.model.enums;

import com.fasterxml.jackson.annotation.JsonCreator;

import jakarta.persistence.EnumeratedValue;

public enum TipoPedido {
    SIMPLES(1),
    DUPLO(2),
    TRIPLO(3);

    @EnumeratedValue
    int value;

    @JsonCreator
    public static TipoPedido fromValue(String value) {
        Integer parsed = Integer.valueOf(value);
        for (TipoPedido t : values()) {
            if (parsed.equals(t.value)) return t;
        }
        throw new IllegalArgumentException("TipoPedido inválido: " + value);
    }

    private TipoPedido(Integer value) {
        this.value = value;
    }

    public Integer getValue() { return value; }
}

File: frontend/planta-smart-4.0/src/domain/enums/TipoPedido.ts
export enum TipoPedido {
  Simples,
  Duplo,
  Triplo,
}

export const TipoPedidoToInt: Record<TipoPedido, number> = {
  [TipoPedido.Simples]: 1,
  [TipoPedido.Duplo]: 2,
  [TipoPedido.Triplo]: 3,
};

export const IntToTipoPedido: Record<number, TipoPedido> = {
  1: TipoPedido.Simples,
  2: TipoPedido.Duplo,
  3: TipoPedido.Triplo,
};

export const TipoPedidoStringToEnum: Record<string, TipoPedido> = {
  SIMPLES: TipoPedido.Simples,
  DUPLO: TipoPedido.Duplo,
  TRIPLO: TipoPedido.Triplo,
};